src/defs.rs

```diff
@@ -53,6 +57,76 @@ pub struct WorkItem {
     pub name: String,
 }
 
+#[derive(Debug)]
```

Comment:
If this stuff is only used in producer.rs, you can define everything there (`defs.rs` is for the shared stuff).